cmd/cstor-pool-mgmt/controller/backup-controller/new_backup_controller.go

```diff
@@ -0,0 +1,232 @@
+			// Note : In case controller reboots and existing object in etcd can cause delivery of
+			// add event when the controller comes again. Be careful in this part and handle accordingly.
+			bkp := obj.(*apis.BackupCStor)
+
+			if !IsRightCStorPoolMgmt(bkp) {
+				return
+			}
+
+			q.Operation = common.QOpAdd
+			glog.Infof("BackupCStor event added: %v, %v", bkp.ObjectMeta.Name, string(bkp.ObjectMeta.UID))
+			controller.recorder.Event(bkp, corev1.EventTypeNormal, string(common.SuccessSynced), string(common.MessageCreateSynced))
+			controller.enqueueBackupCStor(bkp, q)
+		},
+		UpdateFunc: func(old, new interface{}) {
+			// Note : UpdateFunc is called in following three cases:
+			// 1. When object is updated/patched i.e. Resource version of object changes.
+			// 2. When object is deleted i.e. the deletion timestap of object is set.
+			// 3. After every resync interval.
+			newbkp := new.(*apis.BackupCStor)
+			oldbkp := old.(*apis.BackupCStor)
+			glog.Infof("Received Update for backup:%s", oldbkp.ObjectMeta.Name)
+
+			if !IsRightCStorPoolMgmt(newbkp) {
+				return
+			}
+
+			if newbkp.ResourceVersion == oldbkp.ResourceVersion {
+				return
+			}
+
+			if IsDestroyEvent(newbkp) {
+				q.Operation = common.QOpDestroy
+				glog.Infof("BackupCstor Destroy event : %v, %v", newbkp.ObjectMeta.Name, string(newbkp.ObjectMeta.UID))
+				controller.recorder.Event(newbkp, corev1.EventTypeNormal, string(common.SuccessSynced), string(common.MessageDestroySynced))
+			} else {
+				glog.Infof("BackupCstor Modify event : %v, %v", newbkp.ObjectMeta.Name, string(newbkp.ObjectMeta.UID))
+				q.Operation = common.QOpSync
+				controller.recorder.Event(newbkp, corev1.EventTypeNormal, string(common.SuccessSynced), string(common.MessageModifySynced))
+			}
+			controller.enqueueBackupCStor(newbkp, q)
+		},
+		DeleteFunc: func(obj interface{}) {
+			bkp := obj.(*apis.BackupCStor)
+			if !IsRightCStorPoolMgmt(bkp) {
+				return
+			}
+			glog.Infof("BackupCStor Resource delete event: %v, %v", bkp.ObjectMeta.Name, string(bkp.ObjectMeta.UID))
+		},
+	})
+	return controller
+}
+
+// enqueueBackupCStor takes a BackupCStor resource and converts it into a namespace/name
+// string which is then put onto the work queue. This method should *not* be
+// passed resources of any type other than BackupCStor.
+func (c *BackupController) enqueueBackupCStor(obj *apis.BackupCStor, q common.QueueLoad) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	q.Key = key
+	c.workqueue.AddRateLimited(q)
+}
+
+// cleanupOldBackup set fail status to old pending backup
+func cleanupOldBackup(clientset clientset.Interface) {
+	bkplabel := "cstorpool.openebs.io/uid=" + os.Getenv(string(common.OpenEBSIOCStorID))
+	bkplistop := metav1.ListOptions{
+		LabelSelector: bkplabel,
+	}
+	bkplist, err := clientset.OpenebsV1alpha1().BackupCStors(metav1.NamespaceAll).List(bkplistop)
+	if err != nil {
```

Comment:
lets log the error.. and we should take care of handling different types of errors differently

Reply:
ok